### PR TITLE
add replica charge mode

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go [hunk 1]
```diff
@@ -156,6 +159,18 @@ func ResourceRdsReadReplicaInstance() *schema.Resource {
...
+			"charging_mode": common.SchemaChargingMode(nil),
+			"period_unit":   common.SchemaPeriodUnit(nil),
+			"period":        common.SchemaPeriod(nil),
+			"auto_renew": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"true", "false",
+				}, false),
+			},
```
Comment (Lance52259) [inline huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: we have `common.SchemaAutoRenewUpdatable()` method.

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go [hunk 2]
```diff
@@ -196,10 +229,37 @@ func resourceRdsReadReplicaInstanceCreate(ctx context.Context, d *schema.Resourc
...
+			Timeout:      d.Timeout(schema.TimeoutCreate),
+			Delay:        20 * time.Second,
+			PollInterval: 10 * time.Second,
+			// Ensure that the instance is 'ACTIVE', not going to enter 'BACKING UP'.
+			ContinuousTargetOccurence: 2,
+		}
+		if _, err = stateConf.WaitForState(); err != nil {
+			return diag.Errorf("error waiting for replica instance (%s) creation completed: %s", instanceID, err)
+		}
+	}
```
Comment (Lance52259) [inline huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: pls using `common.WaitOrderResourceComplete()` method to instead, and bring into at end of following block:
```
if resp.OrderId != "" {
 ...
}
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go [hunk 3]
```diff
@@ -156,6 +159,18 @@ func ResourceRdsReadReplicaInstance() *schema.Resource {
 				},
 			},
 
+			// charge info: charging_mode, period_unit, period, auto_renew, auto_pay
```
Comment (Lance52259) [inline huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: No `auto_pay` comment

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go [hunk 4]
```diff
@@ -156,6 +158,12 @@ func ResourceRdsReadReplicaInstance() *schema.Resource {
 				},
 			},
 
+			// charge info: charging_mode, period_unit, period, auto_renew
```
Comment (hmbig2) [inline huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go:159]: add these parameters in the document

Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go:159]: done

File: docs/resources/rds_read_replica_instance.md [hunk 5]
```diff
@@ -50,6 +50,10 @@ resource "huaweicloud_rds_read_replica_instance" "replica_instance" {
   volume {
     type = "ULTRAHIGH"
   }
+  charging_mode = "prePaid"
+  period_unit = "month"
+  period = 1
+  auto_renew = "true"
```
Comment (hmbig2) [inline docs/resources/rds_read_replica_instance.md]: pls format this block

File: docs/resources/rds_read_replica_instance.md [hunk 6]
```diff
@@ -81,6 +85,20 @@ The following arguments are supported:
...
+* `period_unit` - (Optional, String, ForceNew) Specifies the charging period unit of the RDS DB instance. Valid values
+  are *month* and *year*. This parameter is mandatory if `charging_mode` is set to *prePaid*. Changing this creates a
+  new resource.
+
+* `period` - (Optional, Int, ForceNew) Specifies the charging period of the RDS DB instance. If `period_unit` is set
+  to *month*, the value ranges from 1 to 9. If `period_unit` is set to *year*, the value ranges from 1 to 3. This
+  parameter is mandatory if `charging_mode` is set to *prePaid*. Changing this creates a new resource.
+
+* `auto_renew` - (Optional, String) Specifies whether auto renew is enabled. Valid values are "true" and "
+  false". Changing this creates a new resource.
```
Comment (Lance52259) [inline docs/resources/rds_read_replica_instance.md]: please remove this: `Changing this creates a new resource.`

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go [hunk 7]
```diff
@@ -196,10 +221,25 @@ func resourceRdsReadReplicaInstanceCreate(ctx context.Context, d *schema.Resourc
...
 	instanceID := d.Id()
-	if err := checkRDSInstanceJobFinish(client, resp.JobId, d.Timeout(schema.TimeoutCreate)); err != nil {
-		return diag.Errorf("error creating replica instance (%s): %s", instanceID, err)
+	// wait for order success
+	if resp.OrderId != "" {
+		bssClient, err := config.BssV2Client(config.GetRegion(d))
+		if err != nil {
+			return diag.Errorf("error creating BSS V2 client: %s", err)
+		}
+		if err := orders.WaitForOrderSuccess(bssClient, int(d.Timeout(schema.TimeoutCreate)/time.Second), resp.OrderId); err != nil {
```
Comment (hmbig2) [inline huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: s/orders.WaitForOrderSuccess/common.WaitOrderComplete

File: docs/resources/rds_read_replica_instance.md (hunk 5 above)
Comment (houpeng80) [reply docs/resources/rds_read_replica_instance.md]: done

File: docs/resources/rds_read_replica_instance.md (hunk 6 above)
Comment (houpeng80) [reply docs/resources/rds_read_replica_instance.md]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

File: docs/resources/rds_read_replica_instance.md [hunk 8]
```diff
@@ -47,6 +47,10 @@ resource "huaweicloud_rds_read_replica_instance" "replica_instance" {
   primary_instance_id   = huaweicloud_rds_instance.instance.id
   availability_zone     = "{{ availability_zone }}"
   enterprise_project_id = "{{ enterprise_project_id }}"
+  charging_mode         = "prePaid"
+  period_unit           = "month"
+  period                = 1
+  auto_renew            = "true"
```
Comment (ShiChangkuo) [inline docs/resources/rds_read_replica_instance.md]: generally, the prepaid charging mode is not shown in basic usage.

File: docs/resources/rds_read_replica_instance.md [hunk 9]
```diff
@@ -81,6 +85,19 @@ The following arguments are supported:
 * `enterprise_project_id` - (Optional, String, ForceNew) The enterprise project id of the read replica instance.
   Changing this parameter will create a new resource.
 
+* `charging_mode` - (Optional, String, ForceNew) Specifies the charging mode of the RDS DB instance. Valid values are
```
Comment (ShiChangkuo) [inline docs/resources/rds_read_replica_instance.md]: s/`RDS DB instance`/`read replica instance`

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go [hunk 10]
```diff
@@ -300,6 +344,17 @@ func resourceRdsReadReplicaInstanceUpdate(ctx context.Context, d *schema.Resourc
 	return resourceRdsReadReplicaInstanceRead(ctx, d, meta)
 }
 
+func updateRdsInstanceAutoRenew(d *schema.ResourceData, config *config.Config) error {
+	bssClient, err := config.BssV2Client(config.GetRegion(d))
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: `HasChange` is missing?

File: docs/resources/rds_read_replica_instance.md (hunk 8 above)
Comment (houpeng80) [reply docs/resources/rds_read_replica_instance.md]: done

File: docs/resources/rds_read_replica_instance.md (hunk 9 above)
Comment (houpeng80) [reply docs/resources/rds_read_replica_instance.md]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go (hunk 10 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_read_replica_instance.go]: done

**Pull request merged**